halo2-base/src/gates/flex_gate.rs

```diff
@@ -426,15 +556,17 @@ pub trait GateInstructions<F: ScalarField> {
         c: impl Into<QuantumCell<F>>,
     ) -> AssignedValue<F>;
 
-    /// assume bits has boolean values
-    /// returns vec[idx] with vec[idx] = 1 if and only if bits == idx as a binary number
+    /// Constrains and returns and indicator vector, where `vec[idx] = 1` iff `idx` equals the binary number represented by `bits` otherwise returns a [Vec] of zeros.
```

Comment:
'and indicator' - typo

It should never return a vector of all zeros. 

It's vec[idx] = 1 iff idx = bits, otherwise vec[idx] = 0.